build/scripts/DevEnv/NewDevEnv.psm1

```diff
@@ -221,19 +129,20 @@ function BuildApp {
     $appFile = $appInfo.GetAppFileName()
 
     if((Test-Path (Join-Path $appOutputFolder $appFile)) -and (-not $rebuild)) {
-        Write-Host "App $appFile already exists in $appOutputFolder. Skipping..."
+        Write-Host "App $appFile already exists in $appOutputFolder. Removing..."
         $appFile = (Join-Path $appOutputFolder $appFile -Resolve)
-    } else {
-        # Create compiler folder on demand
-        if(-not $compilerFolder.Value) {
-            Write-Host "Creating compiler folder..." -ForegroundColor Yellow
-            $compilerFolder.Value = CreateCompilerFolder -packageCacheFolder $packageCacheFolder
-            Write-Host "Compiler folder: $($compilerFolder.Value)" -ForegroundColor Yellow
-        }
+        Remove-Item -Path $appFile -Force
```

Comment:
You're right.. I changed it because the existing implementation didn't recompile e.g. the System App even if I wanted it to be rebuilt. I noticed there is a $rebuild variable in the existing implementation but it's never set anywhere. Perhaps I can leverage that. 

I just want to make sure we recompile e.g. the System App if I run `.\build\scripts\DevEnv\NewDevEnv.ps1 -ProjectPaths '.\src\System Application\App'` even if it was already previously compiled. 

```
if((Test-Path (Join-Path $appOutputFolder $appFile)) -and (-not $rebuild)) {
        Write-Host "App $appFile already exists in $appOutputFolder. Removing..."
        $appFile = (Join-Path $appOutputFolder $appFile -Resolve)
        Remove-Item -Path $appFile -Force
    }
```

Reply:
What if `.\build\scripts\DevEnv\NewDevEnv.ps1 -ProjectPaths '.\src\System Application\*'` fails on building the last app. Do you expect re-running the script to re-compile all apps?
Same goes for re-publishing the apps.

Reply:
Changed this behaviour now such that we'll try to use the cached apps if they exist. If you don't want to used the cached apps you should add the -RebuildApps flag to NewDevEnv.ps1